lstchain/mc/sensitivity.py

```diff
@@ -272,15 +272,11 @@ def calculate_sensitivity_lima(n_on_events, n_background, alpha):
         n_off=n_background,
         alpha=alpha
         )
-
-
     n_excesses_5sigma = stat.excess_matching_significance(5)
-    for value_excesses, value_bkg in np.nditer([n_excesses_5sigma, n_background], op_flags=['readwrite']):
-        if value_excesses < 10:
-            value_excesses[...] = 10
-        if value_excesses < 0.05 * value_bkg * alpha[0]:
-            value_excesses[...]=0.05 * value_bkg * alpha[0]
-        
+    n_excesses_5sigma[n_excesses_5sigma<10] = 10
+    bkg_5percent = 0.05*n_background*alpha
+    n_excesses_5sigma[n_excesses_5sigma<bkg_5percent] = bkg_5percent[n_excesses_5sigma<bkg_5percent]
```

Comment:
same as above. You increase the signal amount to match the requirement. That is wrong!

Reply:
Well,  in the the way we were approaching this, the explanation is that to calculate the sensitivity, we need to know the _minimum signal needed_ in each bin to reach 5 sigma significance. This minimum signal (minimum number of excesses over background) is calculated by the Li&Ma function, but for some bins these calculated excesses are too low, so the **actual** signal needed would be higher (at least 10, or at least 5% of the background). Its not that we are increasing the amount of signal, but we are increasing the amount of signal _required_ in those bins.

Reply:
I have not checked the full code, but it seems ok to me, by increasing the number you will anyway "over-fulfill" the other conditions. Then, the larger number of gammas will translate in a higher flux (=worse sensitivity).

Reply:
I think we can mark this as resolved, please consider changing this in `pyirf` if it was implemented in the way you thought was right @maxnoe